lib/cocoapods/generator/copy_xcframework_script.rb

```diff
@@ -158,7 +162,7 @@ def script
   local target_path="$SELECT_SLICE_RETVAL"
   if [[ -z "$target_path" ]]; then
     echo "warning: [CP] Unable to find matching .xcframework slice in '${paths[@]}' for the current build architectures ($ARCHS)."
-    return
+    exit 1
```

Comment:
if we're changing this we should also change the above `warning:` to `error:` so it shows a proper error in Xcode